SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp

```diff
@@ -378,9 +378,13 @@ void USpatialSender::CreateServerWorkerEntity(int AttemptCounter)
 
 		UE_LOG(LogSpatialOSNetDriver, Warning, TEXT("Worker entity creation request timed out and will retry."));
 		FTimerHandle RetryTimer;
-		TimerManager->SetTimer(RetryTimer, [this, AttemptCounter]()
+		TWeakObjectPtr<USpatialSender> WeakThis(this);
```

Comment:
I'm not sure why I couldn't define this inside the lambda like the other usages. I'll try to see why the compiler is unhappy but wanted to push this out to get a review of the other changes.

Reply:
I think this is a compiler bug where if `this` is used in the constructor for `TWeakObjectPtr` inside the capture the compiler thinks it points to the lambda instead of to an instance of `USpatialSender`.

Reply:
Hmm Yes that sounds likely. Cant you create it in the outmost lambda and then recapture it here rather than creating the weakpointer already inside a lambda?

Reply:
Is this resolved? +1 to dan's comment to create the weakptr outside the lambda and pass in?

Reply:
This works fine - Dan agreed the outer lambda doesn't need to capture a weakptr because its invocation depends on the receiver existing so sender should also exist otherwise many other things will break. What I wanted to do - declare the weakptr in the capture list, unfortunately doesn't compile in VS2017 (I think it's because this bug wasn't properly fixed or it's just not released for VS2017 https://developercommunity.visualstudio.com/content/problem/246948/msconnect-3128879-this-in-capture-expression-of-ne.html)